files/dhcp/sethostname

```diff
@@ -1,20 +1,19 @@
 #!/bin/bash
-# This script is to update hostname of the system.
 
-if [ "$reason" != BOUND ] && [ "$reason" != RENEW ] \
-     && [ "$reason" != REBIND ] && [ "$reason" != REBOOT ]
+if [ "$reason" = BOUND ]  &&  ["$reason" != RENEW ] \
+   && ["$reason" != REBIND ]  &&  ["$reason" != REBOOT ]
 then
     exit 0
 fi
 
 current_host_name=`hostname -s`
 
-if [ "$current_host_name" != "$new_host_name" ]
+if [ "$current_host_name" != "$new_host_name" ] && [ ! -z "$new_host_name" ]
 then
     echo $new_host_name > /etc/hostname
-    line_to_replace=`grep 127.0.0.1.*$current_host_name /etc/hosts`
-    new_line=`echo $line_to_replace | sed "s/$current_host_name/$new_host_name/"`
-    sed -i "s/$line_to_replace/$new_line/" /etc/hosts
-
     hostname -F /etc/hostname
+    sed -i "/\s$current_host_name$/d" /etc/hosts
+    echo "127.0.0.1 $new_host_name" >> /etc/hosts
 fi
```

Comment:
From observation this bash will be called twice in a row sometimes, "oldhostname" -> "" and "" -> "newhostname". Old implementation is not working correctly under these cases.

---
In reply to: [100569696](https://github.com/Azure/sonic-buildimage/pull/288#discussion_r100569696) [](ancestors = 100569696)